external-crates/move/crates/move-core-types/src/annotated_value.rs

```diff
@@ -410,24 +409,29 @@ impl fmt::Display for MoveValue {
             MoveValue::Bool(true) => write!(f, "true"),
             MoveValue::Address(a) => write!(f, "{}", a.to_hex_literal()),
             MoveValue::Signer(a) => write!(f, "signer({})", a.to_hex_literal()),
-            MoveValue::Vector(v) => fmt_list(f, "vector[", v, "]"),
+            MoveValue::Vector(v) => {
+                use DebugAsDisplay as DD;
+                write!(f, "vector")?;
+                let mut list = f.debug_list();
+                for val in v {
+                    list.entry(&DD(val));
+                }
+                list.finish()
+            }
             MoveValue::Struct(s) => fmt::Display::fmt(s, f),
         }
     }
 }
 
 impl fmt::Display for MoveStruct {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use DebugAsDisplay as DD;
         fmt::Display::fmt(&self.type_, f)?;
-        fmt_list(f, " {", self.fields.iter().map(DisplayFieldBinding), "}")
-    }
-}
-
-struct DisplayFieldBinding<'a>(&'a (Identifier, MoveValue));
-
-impl fmt::Display for DisplayFieldBinding<'_> {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let DisplayFieldBinding((field, value)) = self;
-        write!(f, "{}: {}", field, value)
+        write!(f, " ")?;
```

Comment:
What's this space do?

Reply:
So that in a struct you get `0x2::bar::foo { ... }` instead of `0x2::bar::foo{ ... }`